File: Web_Programming/Web Programming 1/JavaScript/Category.js
/*
	Class 7
	9/21/15
*/

function Category(aCategoryId, aCategoryName, aDescription)
{
	this.CategoryId = aCategoryId;
	this.CategoryName = aCategoryName;
	this.Description = aDescription;
	
}

Category.prototype.toString = function()
{
	var output = "CategoryId = " + this.CategoryId + "<br>";
		output = output + "CategoryName = " + this.CategoryName + "<br>";
		output = output + "Description = " + this.Description + "<br>";
		return output;
	
}
//	This shares the method among instances
//	instead of having a different function 
//	for eachinstance

function Shipper(aId, aName, aPhone)
{
	this.ShipperId = aId;
	this.Name = aName;
	this.Phone = aPhone;
	
}

Shipper.prototype.toString = function()
{
	var out = "ShipperId " + this.ShipperId + "<br>";
	out = out + "Shipper Name " + this.Name + "<br>";
	out = out + "Shipper Phone " + this.Phone + "<br>";
	return out;
}	


/*
	Shipper.prototype.ShipperId = 1;
	Shipper.prototype.Name = "Speedy Shippers";
	Shipper.prototype.Phone = "(123) 867-5309";
	Shipper.prototype.toString = function()
	{
		var out = "ShipperId " + this.ShipperId + "<br>";
		out = out + "Shipper Name " + this.Name + "<br>";
		out = out + "Shipper Phone " + this.Phone + "<br>";
		return out;
	}

*/

var category1 = new Category(1, "Beverages", "Soft drinks, coffee, teas, beers, and ales");
var category2 = new Category(2, "Condiments", "Sweet and savory sauces, relishes, spreads, and seasonings");

var results = document.getElementById("results");

/*
var result = category1.CategoryId + " ";
result = result + category1.CategoryName + " ";
result = result + category1.Description + "<br><br>";

result = result + category2.CategoryId + " ";
result = result + category2.CategoryName + " ";
result = result + category2.Description + "<br><br>";

result = result + "<br><br><br><br><br>";
*/

var result = category1.toString() + "<br> ";
result += category2.toString() + "<br>";

result += "category1 is an instance of Category " + (category1 instanceof Category) + " <br>";
result += "category2 is an instance of Category " + (category2 instanceof Category) + " <br>";

result += "category1 is an instance of Object " + (category1 instanceof Object) + " <br>";
result += "category2 is an instance of Object " + (category2 instanceof Object) + " <br>";
result += "<br><br><br>";

result += "category1.toString is the same as category2 toString " 
			+ (category1.toString == category2.toString) + "<br>";

			
var shipper1 = new Shipper(1, "Speedy Express", "(503) 555-9831");
var shipper2 = new Shipper(2, "United Package", "(503) 555-3199");

//	In C# or Java this would give you an error
//	But in JavaScript the prototype variable is masked 
//	by the instance 
//	Don't DO THIS!!!!
// shipper1.Name = "FedEx";

result += "<br><br><br><br><br>";

result += shipper1.toString();
result += shipper2.toString();

result += "shipper1.toString is the same as shipper2 toString "
	+ (shipper1.toString == shipper2.toString) + "<br><br>";
	
	
	
results.innerHTML = result;

File: Web_Programming/Web Programming 1/Northwind4/Northwind4/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Text;
using System.Threading.Tasks;

namespace Northwind4.Models
{

    public class Category
    {
        private int categoryId = -1;
        private string categoryName = "n/a";
        private string description = "n/a";


        public int CategoryId
        {
            get
            {
                return this.categoryId;
            }
            set
            {
                this.categoryId = value;

            }


        }

        public string CategoryName
        {
            get
            {
                return this.categoryName;
            }

            set
            {
                this.categoryName = value;
            }
        }

        public string Description
        {
            get
            {
                return this.description;
            }

            set
            {
                this.description = value;
            }
        }

		// Constructors
		public Category()
		{

		}

		public Category(int aCategoryId, string aCategoryName, string aDescription)
			: this()
		{
			this.CategoryId = aCategoryId;
			this.CategoryName = aCategoryName;
			this.Description = aDescription;

		}

		public Category(int aCategoryId, string aCategoryName)
			: this(aCategoryId, aCategoryName, "n/a")
		{

		}

        

        // ToString  

        public override string ToString()
        {
			string aString = "";
			aString += "Category ID = " + this.CategoryId + "<br />";
			aString += "Category Name = " + this.CategoryName + "<br />";
			aString += "Description = " + this.Description + "<br /><br />";
			           
            return aString;
        }

        public string Display()
        {
            string aString = "";
            aString += "Category ID = " + CategoryId + "<br />";
            aString += "Category Name = " + CategoryName + "<br />";
            aString += "Description = " + Description + "<br /><br />";
           
            return aString;
        }
    }
}